lib/server/services/qualityControlFlag/QcFlagTypeService.js

```diff
@@ -54,6 +60,50 @@ class QcFlagTypeService {
         return qcFlag;
     }
 
+    /**
+     * Create new QC Flag Type
+     * @param {object} [parameters] qc flag parameters
+     * @param {string} [parameters.name] qc flag name
+     * @param {string} [parameters.method] qc flag method
+     * @param {boolean} [parameters.bad] qc flag bad attribute
+     * @param {string} [parameters.color] qc flag color
+     * @param {number} [parameters.userId] qc flag creator id
+     * @param {number} [parameters.externalUserId] qc flag creator external id
+     * @return {Promise<QCFlagType>} promise
+     */
+    async create(parameters) {
+        const {
+            name,
+            method,
+            bad,
+            color,
+            userId,
+            externalUserId,
+        } = parameters;
+
+        const user = await getUserOrFail({ userId, externalUserId });
+        if (color && !/#[0-9a-fA-F]{6}/.test(color)) {
+            throw new BadParameterError(`Incorrect format of the color provided (${color})`);
+        }
```

Comment:
Shouldn't this be tested in dto? If it's too complex I'm ok leaving this here

Reply:
right, it can be, it's small changed so moved to controller